### DISCUSSION TITLE
Differences in beatmap selection on game-open and search

### DISCUSSION THREAD

**Discussion by Bubbleman532** [ppy/osu#28000] [Ideas]
I'd like to open a discussion about how the lazer client handles beatmap selection on open and on search.

In stable, if the themed music option is turned off, a random beatmap plays on opening the client, which when the player enters song select is selected by default. Lazer behaviour doesn't match this, instead selecting another random song. A couple times I've wanted to play the map that plays when the game loads and have had it switched on me without remembering which song was playing.

Also, on searching in stable, the previous beatmap stays selected whilst typing until only one beatmap matches the filter, whereas in lazer the map switches immediately when the previous beatmap no longer matches the filter.

I'm wondering if this stable behaviour could be replicated, though if the new behaviour is intentional I'm open to hearing why.

**Comment by SupDos** [comment]
> Lazer behaviour doesn't match this, instead selecting another random song

Just a small clarification on your first point, it is different to stable, but in a different way

It's possible that the random beatmap it selected was for a different ruleset than the one you currently have selected (for example, you have standard selected, and the beatmap it chose is a taiko beatmap), so when you go into song select, it picks a different standard beatmap.

You can replicate this by picking a map in one ruleset, going back to the main menu, switching rulesets then going back into song select (make sure show converted is turned off).

This song select mechanic is different to stable: in stable it just shows *every beatmap in your song select* regardless of what ruleset you have selected, so no matter what song it picks to play it will jump to it.

**Reply by Bubbleman532** [reply]
Good correction, thank you

**Comment by peppy** [comment]
I wouldn't be opposed to limiting main menu music playback to only beatmaps for the current ruleset, which would fix this.

**Reply by rednir** [reply]
Allowing a beatmap that is not visible with the current filters to be selected would also fix this issue (along with other annoyances, like this https://github.com/ppy/osu/discussions/14564, and the awkward silence when no maps are visible, and just generally matching stable behaviour which is what people are used to)

**Reply by peppy** [reply]
Yeah, that's also a valid direction. Will see if we can easily make that happen.

**Comment by WitherFlower** [comment]
> Also, on searching in stable, the previous beatmap stays selected whilst typing until only one beatmap matches the filter, whereas in lazer the map switches immediately when the previous beatmap no longer matches the filter.

I fully support this, and would like to add that this becomes really frustrating if you accidentally press a key on your keyboard, causing your map to disappear and another one to be selected potentially hundreds/thousands of items away from the one you were previously on.

**Comment by Maxwell-0** [comment]
I have another suggestion that is somewhat related: As far as I'm aware the 'Show converted' checkbox in Lazer doesn't actually do anything while the basic Osu ruleset is selected, correct? In that case, I would suggest swapping it for a 'Show all' checkbox (only for basic Osu), which lets you browse all beatmaps regardless of ruleset. This should match the song select in stable when selecting basic Osu ruleset.

Additionally, I like to use Osu main menu as a music player. Osu Lazer is slightly limited in this regard as it doesn't randomize the songs yet, but I think that is supposed to change eventually? So, at least when 'Show all' is selected for Osu ruleset, I would like to have all the songs available for playback in the main menu, ideally with the random feature which will hopefully be added some day.

**Reply by SupDos** [reply]
osu isn't a music player, you're only going to make the devs care less about your suggestions if you try and suggest that
